arwen/host/arwen.go

```diff
@@ -60,7 +60,7 @@ type vmHost struct {
 // NewArwenVM creates a new Arwen vmHost
 func NewArwenVM(
 	blockChainHook vmcommon.BlockchainHook,
-	vmExecutor executor.Executor,
```

Comment:
As already discussed, `NewArwenVM()` must not receive any extra arguments. It must create the executor / factory on its own. It must own both the executor and the factory.

Otherwise, the executor / factory must be initialized and owned by an outside component, which must now know details about the low-level executor itself. It must not come to that.

Add necessary fields to `VMHostParameters` if needed; it belongs to the VM and can be freely modified. For example, add a field called `AlternativeExecutorFactory` which is used by `NewArwenVM()` if it's not `nil`. Otherwise, if `nil`, simply instantiate a `wasmer.ExecutorFactory` _**within NewArwenVM()**_.